taskcluster/docker/node/build.py

```diff
@@ -75,17 +75,32 @@ def write_package_info(package_info):
 
 def get_buildid():
     now = datetime.utcnow()
-    return now.strftime("%Y%m%d.%H%M%S")
+    # note the `-` (hyphen) in `%-H`
+    # this removes the leading zero
+    # from the hour (leading zeros are not allowed)
+    return now.strftime("%Y%m%d.%-H%M%S")
 
 
 def get_buildid_version(version):
-    """Version schema check+append a `buildid{buildid}` to ensure unique version."""
-    # XXX is there a more precise schema we should verify ourselves against?
-    if len(version.split(".")) not in (1, 2, 3):
-        raise Exception("{version} has too many version parts!")
-    if "buildid" in version:
-        raise Exception("{version} already has a buildid specified!")
-    buildid_version = f"{version}buildid{get_buildid()}"
+    """Check the version's formating and append `date.time` as a buildid to ensure a unique version.
+    The addon's in-tree `package.json` file specifies `major.minor.0`.
+    The format of the resulting version field is:
+        <number>.<number>.%Y%m%d.%-H%M%S
+    """
+    parts = version.split(".")
+    if len(parts) not in (1, 2, 3):
```

Comment:
Shouldn't this be (1,2,3,4)? From the docs:
> The version string consists of 1 to 4 numbers separated by dots, for example, 1.2.3.4. Non-zero numbers must not include a leading zero. For example, 2.01 is not allowed; however, 0.2, 2.0.1, and 2.10 are allowed.

Reply:
Elements 3 and 4 are over-ridden by the pipeline to append a timestamp in such a way that the generated versions are compatible with MV3. Ideally it would be `(1, 2)` but two part versions break a library called [node-semver](https://github.com/npm/node-semver). Two part versions are technically valid in the semver spec, [but the version in `package.json` must be parasable by `node-semver` to be a valid semantic version as far as `npm` is concerned](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#version). This library thinks a version must have 3 parts to be a valid semantic version (and tooling relying on this library breaks when `package.json` has a version that this library can't parse 🫠)

Reply:
So allowing a third part (that is over-ridden) is a escape hatch for the addons relying on such tooling.

Reply:
Actually, taking a look at the spec again it sounds like it requires a 3 part version.